### PR TITLE
Build docker image

### PR DESCRIPTION
**Pull request by TrevCraw**
https://github.com/OpenLiberty/ci.maven/issues/792

### CODE REVIEW

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java [hunk 1]
```diff
@@ -317,6 +321,8 @@ public DevUtil(File serverDirectory, File sourceDirectory, File testSourceDirect
             this.trackingMode = FileTrackMode.NOT_SET;
         }
         this.container = container;
+        this.imageName = "open-liberty:latest";
```
[the remark this answers is not in the document]
Comment (TrevCraw) [reply src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java]: Updated

File: src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java [hunk 2]
```diff
@@ -281,16 +282,19 @@ public abstract boolean recompileBuildFile(File buildFile, List<String> artifact
...
             List<File> resourceDirs, boolean hotTests, boolean skipTests, boolean skipUTs, boolean skipITs,
             String applicationId, long serverStartTimeout, int appStartupTimeout, int appUpdateTimeout,
             long compileWaitMillis, boolean libertyDebug, boolean useBuildRecompile, boolean gradle, boolean pollingTest,
-            boolean container) {
+            boolean container, File dockerfile) {
         this.serverDirectory = serverDirectory;
         this.sourceDirectory = sourceDirectory;
         this.testSourceDirectory = testSourceDirectory;
         this.configDirectory = configDirectory;
+        this.projectBaseDir = (sourceDirectory == null) ? null : sourceDirectory.getParentFile().getParentFile().getParentFile(); // {dir}/src/main/java <---
```
[the remark this answers is not in the document]
Comment (TrevCraw) [reply src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java:298]: @turkeylurkey will be opening a PR that will deal with this. The plan is to merge this PR after his.

**Comment by TrevCraw** [comment]
@turkeylurkey I went with the Dockerfile parent as the build context, because in typical Docker usage, it is assumed the Dockerfile is located in the build context path. It is possible that we should allow the build context to be an option for the user to specify. @ericglau what are your thoughts on this?

**Comment by ericglau** [comment]
We could consider exposing the docker build path (build context) as another dev mode parameter, since it is a required parameter for the docker build command.  Default should be the Dockerfile path as it currently is.

**Comment by TrevCraw** [comment]
Opened issue for handling docker build context here: https://github.com/OpenLiberty/ci.maven/issues/830

**Pull request merged**